src/main/java/org/apache/sling/resourceresolver/impl/VanityPathConfigurer.java

```diff
@@ -0,0 +1,142 @@
+            }
+            Collections.sort(this.vanityPathConfig);
+        }
+    }
+
+    public int getDefaultVanityPathRedirectStatus() {
+        return config.resource_resolver_default_vanity_redirect_status();
+    }
+
+    public boolean isVanityPathEnabled() {
+        return this.config.resource_resolver_enable_vanitypath();
+    }
+
+    public boolean isVanityPathCacheInitInBackground() {
+        return this.config.resource_resolver_vanitypath_cache_in_background();
+    }
+
+    public List<VanityPathConfig> getVanityPathConfig() {
+        return this.vanityPathConfig;
+    }
+
+
+    public boolean hasVanityPathPrecedence() {
+        return this.config.resource_resolver_vanity_precedence();
+    }
+
+    public long getMaxCachedVanityPathEntries() {
+        return this.config.resource_resolver_vanitypath_maxEntries();
+    }
+
+    public boolean isMaxCachedVanityPathEntriesStartup() {
+        return this.config.resource_resolver_vanitypath_maxEntries_startup();
+    }
+
+    public int getVanityBloomFilterMaxBytes() {
+        return this.config.resource_resolver_vanitypath_bloomfilter_maxBytes();
+    }
+
+    private boolean isDefined(final String[] value) {
+        return value != null && value.length > 0;
+    }
+
+    List<String> configureVanityPathPrefixes(final String[] pathPrefixes, final String[] pathPrefixesFallback,
+                                     String pathPrefixesPropertyName, String pathPrefixesFallbackPropertyName) {
+        if (isDefined(pathPrefixes) && isDefined(pathPrefixesFallback) ) {
+            logger.error("Both properties, " + pathPrefixesPropertyName + " and " + pathPrefixesFallbackPropertyName
+                + ", were defined. Using " + pathPrefixesPropertyName + " for configuring vanity paths. "
+                + "Please remove the other property from your configuration.");
+            return filterVanityPathPrefixes(pathPrefixes);
+        } else if (isDefined(pathPrefixes)) {
+            return filterVanityPathPrefixes(pathPrefixes);
+        } else if (isDefined(pathPrefixesFallback)) {
+            logger.warn("The property " + pathPrefixesPropertyName + " was not set. Using the " +
+                pathPrefixesFallbackPropertyName + " instead. Please update your configuration to use " + pathPrefixesPropertyName);
+            return filterVanityPathPrefixes(pathPrefixesFallback);
+        }
+        return null;
+    }
+
+    private static List<String> filterVanityPathPrefixes(final String[] vanityPathPrefixes) {
+        final List<String> prefixList = new ArrayList<>();
+        for (final String value : vanityPathPrefixes) {
+            if (value.trim().length() > 0) {
+                if (value.trim().endsWith("/")) {
+                    prefixList.add(value.trim());
+                } else {
+                    prefixList.add(value.trim().concat("/"));
+                }
+            }
+        }
+        return prefixList.isEmpty() ? null : prefixList;
```

Comment:
I bet this could look nicer with `Stream`s and `Optional`! But, of course, that's a matter of taste.